From a4147e19f3f2620d92bbdb6ffc940f2ea06f808d Mon Sep 17 00:00:00 2001
From: Sachin Prabhu <sprabhu@redhat.com>
Date: Mon, 28 Jul 2014 12:16:48 -0400
Subject: [fs] cifs: fix dead code

Message-id: <1406549822-10376-56-git-send-email-sprabhu@redhat.com>
Patchwork-id: 86743
O-Subject: [RHEL7.1 PATCH 55/69] [bz1082049] cif: fix dead code
Bugzilla: 1082049
RH-Acked-by: J. Bruce Fields <bfields@redhat.com>
RH-Acked-by: Steve Dickson <SteveD@redhat.com>

From: Michael Opdenacker <michael.opdenacker@free-electrons.com>

This issue was found by Coverity (CID 1202536)

This proposes a fix for a statement that creates dead code.
The "rc < 0" statement is within code that is run
with "rc > 0".

It seems like "err < 0" was meant to be used here.
This way, the error code is returned by the function.

Signed-off-by: Michael Opdenacker <michael.opdenacker@free-electrons.com>
Acked-by: Al Viro <viro@zeniv.linux.org.uk>
Signed-off-by: Steve French <smfrench@gmail.com>
(cherry picked from commit 1f80c0cc39e587edd06a36b43ba3a3b09d4ac428)
Signed-off-by: Sachin Prabhu <sprabhu@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/cifs/file.c b/fs/cifs/file.c
index 483271c..e1e3a07 100644
--- a/fs/cifs/file.c
+++ b/fs/cifs/file.c
@@ -2600,7 +2600,7 @@ cifs_writev(struct kiocb *iocb, const struct iovec *iov,
    ssize_t err;
 
    err = generic_write_sync(file, iocb->ki_pos - rc, rc);
-   if (rc < 0)
+   if (err < 0)
     rc = err;
   }
  } else {
-- 
1.7.1